--- git status ---
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks.egg-info/PKG-INFO
	modified:   parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
	modified:   parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	Isaaclab_Parkour.egg-info/
	logs/
	outputs/
	parkour_tasks/parkour_tasks.egg-info/requires.txt
	parkour_tasks/pyproject.toml

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc
index 35791c5..f6f000f 100644
Binary files a/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc
index 42278bf..e0b702f 100644
Binary files a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc and b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc
index 3898f38..70841b6 100644
Binary files a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc and b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc
index ebd5074..4156df3 100644
Binary files a/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc
index 58a4921..cb77b54 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc
index 10aa9c5..6b02cd9 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc
index 18c1f37..57e4a6b 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc
index 0de7e43..2577d18 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc
index cc3960c..520cba3 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc
index 2e880e4..fc149a4 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc
index db2854c..871d904 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc
index 11eea17..4c72b3b 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc
index add777a..256dbc9 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc
index 7c32b5c..fce8bf3 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc
index 2110747..1ec0e43 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc
index 3deac1b..ec4a88c 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc
index 54aba5c..2218f8b 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc
index 5df0572..2af2b58 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc
index 2058977..f224ae6 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc
index ce87cc1..cd16b08 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc
index 5619457..88f7d95 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc
index 1c69be6..fbc5acf 100644
Binary files a/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc
index 0870b87..3eadc5d 100644
Binary files a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc
index 647cd72..a7d4198 100644
Binary files a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc
index b961570..2d07b31 100644
Binary files a/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc
index fff635f..5c8a7c2 100644
Binary files a/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc
index d8053d5..7034ff0 100644
Binary files a/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc
index 505655b..0cf4941 100644
Binary files a/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc
index d38f6df..e9502f6 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc
index 60beae6..903b683 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc
index 1732196..beb8011 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc
index f3fc89f..8a9d96a 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc
index 6d6b74a..dfd1490 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc
index b83a15e..ad0b6a7 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc
index c82f2a8..e2072c1 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc
index 6fd68b3..57fdc66 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc
index 3516132..4115e38 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc
index 8e04828..703676d 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks.egg-info/PKG-INFO b/parkour_tasks/parkour_tasks.egg-info/PKG-INFO
index a55582a..07cbb6c 100644
--- a/parkour_tasks/parkour_tasks.egg-info/PKG-INFO
+++ b/parkour_tasks/parkour_tasks.egg-info/PKG-INFO
@@ -10,10 +10,12 @@ Classifier: Natural Language :: English
 Classifier: Programming Language :: Python :: 3.10
 Classifier: Isaac Sim :: 4.5.0
 Requires-Python: >=3.10
+Requires-Dist: fast_simplification
 Dynamic: author
 Dynamic: classifier
 Dynamic: home-page
 Dynamic: keywords
 Dynamic: maintainer
+Dynamic: requires-dist
 Dynamic: requires-python
 Dynamic: summary
diff --git a/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt b/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
index 4e068b7..95daf85 100644
--- a/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
+++ b/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
@@ -1,3 +1,4 @@
+pyproject.toml
 setup.py
 parkour_tasks/__init__.py
 parkour_tasks/default_cfg.py
@@ -5,4 +6,5 @@ parkour_tasks.egg-info/PKG-INFO
 parkour_tasks.egg-info/SOURCES.txt
 parkour_tasks.egg-info/dependency_links.txt
 parkour_tasks.egg-info/not-zip-safe
+parkour_tasks.egg-info/requires.txt
 parkour_tasks.egg-info/top_level.txt
\ No newline at end of file
diff --git a/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc
index 5c14729..45fcfa7 100644
Binary files a/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc
index c06b1ac..ae73b30 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc
index 65642b7..c37526a 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc
index dca2219..cd4327b 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc
index 7edbd96..bfac375 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc
index 6891ec5..2824d39 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc
index 590be96..185672d 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc
index 371ec55..a8973b0 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc b/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc
index 6acb492..5979e66 100644
Binary files a/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc b/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc
index 03b4f26..f402e96 100644
Binary files a/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc b/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc
index d1dc3e9..761fa04 100644
Binary files a/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc b/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc
index 7ec4a3e..2be3c2e 100644
Binary files a/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc
index dd03936..91a3dcf 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc
index adeed19..b9e73b0 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc
index b09528f..6ce6f3f 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc
index 4503366..bd7030d 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc
index 6e8058b..67b52f5 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc
index 081d346..35deff1 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc
index 376783d..6d9b691 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc
index 72b5b43..7e5f8f0 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc
index aeb2c8b..10baeb4 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc differ